core/domain/stats_domain.py

```diff
@@ -403,6 +405,215 @@ def validate(self):
+
+            exp_domain.validate_customization_args_and_values(
+                'issue', issue['issue_id'], issue['issue_customization_args'],
+                issue_instance.customization_arg_specs)
+
+
+class Playthrough(object):
+    """Domain object representing a learner playthrough.
+    """
+
+    def __init__(
+            self, playthrough_id, exp_id, exp_version, issue_id,
+            issue_customization_args, playthrough_actions, is_valid):
+        """Constructs a Playthrough domain object.
+
+        Args:
+            playthrough_id: str. ID of the playthrough.
+            exp_id: str. ID of the exploration.
+            exp_version: int. Version of the exploration.
+            issue_id: str. ID of the issue.
+            issue_customization_args: dict. The customization args dict for the
+                given issue_id.
+            playthrough_actions: list(dict). The playthrough actions for this
+                playthrough. This will be a list of dicts where each dict
+                represents a single playthrough action. The list is ordered by
+                the time of occurence of the action.
+            is_valid: Bool. Whether the playthrough is valid.
+        """
+        self.id = playthrough_id
+        self.exp_id = exp_id
+        self.exp_version = exp_version
+        self.issue_id = issue_id
+        self.issue_customization_args = issue_customization_args
+        self.playthrough_actions = playthrough_actions
+        self.is_valid = is_valid
+
+    def to_dict(self):
+        """Returns a dict representation of the Playthrough domain object.
+
+        Returns:
+            dict. A dict mapping of all fields of Playthrough object.
+        """
+        return {
+            'id': self.id,
+            'exp_id': self.exp_id,
+            'exp_version': self.exp_version,
+            'issue_id': self.issue_id,
+            'issue_customization_args': self.issue_customization_args,
+            'playthrough_actions': self.playthrough_actions,
+            'is_valid': self.is_valid
+        }
+
+    @classmethod
+    def from_dict(cls, playthrough_dict):
+        """Returns a Playthrough object from a dict.
+
+        Args:
+            playthrough_dict: dict. A dict mapping of all fields of Playthrough
+                object.
+
+        Returns:
+            Playthrough. The corresponding Playthrough domain object.
+        """
+        return cls(
+            playthrough_dict['id'],
+            playthrough_dict['exp_id'],
+            playthrough_dict['exp_version'],
+            playthrough_dict['issue_id'],
+            playthrough_dict['issue_customization_args'],
+            playthrough_dict['playthrough_actions'],
+            playthrough_dict['is_valid'])
+
+    def validate(self):
+        """Validates the Playthrough domain object."""
+        if not isinstance(self.id, basestring):
+            raise utils.ValidationError(
+                'Expected ID to be a string, received %s' % self.id)
```

Comment:
here and everywhere else, shouldn't we use type()?

Reply:
Done.